File: backend/src/services/glucose.service.js
const Glucose = require("../models/Glucose");

exports.addGlucose = async (userId, data) => {
  const { glucoseLevel, measurementType, notes } = data;

  if (!glucoseLevel) {
    throw new Error("Glucose level is required");
  }

  if (!measurementType) {
    throw new Error("Measurement type is required");
  }

  const record = await Glucose.create({
    user: userId,
    glucoseLevel,
    measurementType,
    notes,
  });

  return record;
};

exports.getMyGlucose = async (userId) => {
  return await Glucose.find({ user: userId }).sort({ date: -1 });
};

File: mobile_app/lib/core/services/glucose_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

import 'api_service.dart';
import 'auth_service.dart';

class GlucoseService {
  /// ADD GLUCOSE
  static Future<bool> addGlucose({
    required double glucoseLevel,
    required String measurementType,
    String? date,
    String? notes,
  }) async {
    try {
      final token = await AuthService.getToken();

      final response = await http.post(

        Uri.parse('${ApiService.baseUrl}/glucose/add'),
        headers: {
          'Content-Type': 'application/json',
          'Authorization': 'Bearer $token',
        },
        body: jsonEncode({
          "glucoseLevel": glucoseLevel,
          "measurementType": measurementType,
          "date": date,
          "notes": notes,
        }),
      );
      // delete 2 line after that
      print("GLUCOSE STATUS: ${response.statusCode}");
      print("GLUCOSE BODY: ${response.body}");


      final data = jsonDecode(response.body);

      return (response.statusCode == 200 || response.statusCode == 201) &&
    data['success'] == true;
    } catch (e) {
      return false;
    }
  }

  /// GET MY GLUCOSE HISTORY
  static Future<List<dynamic>> getMyGlucose() async {
    try {
      final token = await AuthService.getToken();

      final response = await http.get(
        Uri.parse('${ApiService.baseUrl}/glucose/my'),
        headers: {
          'Content-Type': 'application/json',
          'Authorization': 'Bearer $token',
        },
      );

      final data = jsonDecode(response.body);

      if (response.statusCode == 200 && data['success'] == true) {
        return data['data'] ?? [];
      }

      return [];
    } catch (e) {
      return [];
    }
  }
}